File: 1122-relative-sort-array/1122-relative-sort-array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
var relativeSortArray = function(arr1, arr2) {
    let h = {};
    for(let i=0;i<arr2.length;++i){
        h[arr2[i]]=i;
    }  
    for(const i of arr1){
        if(!(i in h)) h[i]=1000+i;
    }    
    arr1.sort((a,b)=>h[a]-h[b]);
    return arr1;
};

File: 1122-relative-sort-array/1122-relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        m = {}
        for i,x in enumerate(arr2):
            m[x] = i
        for num in arr1:
            if num not in m:
                m[num] = num + 1000
        sorted_arr = sorted(arr1, key=lambda x: m[x])
        return sorted_arr